java/src/com/ibm/streamsx/rest/Instance.java

```diff
@@ -132,29 +176,75 @@ public static Instance ofEndpoint(String endpoint, String name, String userName,
      */
     public static Instance ofEndpoint(String endpoint, String name, String userName, String password,
             boolean verify) throws IOException {
-        
-        if (name == null)
-            name = Util.getenv(Util.STREAMS_INSTANCE_ID);
-               
+
+        boolean possible_integ = true;
+        if (endpoint == null) {
+            endpoint = System.getenv(Util.ICP4D_DEPLOYMENT_URL);
+            if (endpoint == null) {
+                possible_integ = false;
+                endpoint = System.getenv(Util.STREAMS_REST_URL);
+                if (endpoint == null) {
+                    return null;
+                }
+            }
+        }
+        if (possible_integ && name == null)
+            name = System.getenv(Util.STREAMS_INSTANCE_ID);
+
+        StreamsConnection conn = null;
```

Comment:
Don't initialize to `null` - the value is set in both paths and not setting it to `null` would correctly cause a compile error if it was used before set.